Short-Answer/Algorithms_Answers.md

```diff
@@ -2,13 +2,13 @@
 
 ## Exercise I
 
-a)
+a) O(n) because its running linearly
 
 
-b)
+b) O(n^2) because it contains two loops which are nested
```

Comment:
We multiply the individual runtimes of each loop to obtain the runtime of an entire block since these loops are nested, giving us an overall runtime of O(n log(n)).
```
sum = 0
    for i in range(n):
      j = 1
      while j < n:
        j *= 2
        sum += 1
```